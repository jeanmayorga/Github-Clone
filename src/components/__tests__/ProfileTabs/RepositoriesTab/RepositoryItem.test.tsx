import { render } from '@testing-library/react'
import { RepositoryItem } from '../../../'

export const repositoryMock = {
  id: 158453569,
  node_id: 'MDEwOlJlcG9zaXRvcnkxNTg0NTM1Njk=',
  name: 'midu.dev',
  full_name: 'midudev/midu.dev',
  private: false,
  owner: {
    login: 'midudev',
    id: 1561955,
    node_id: 'MDQ6VXNlcjE1NjE5NTU=',
    avatar_url: 'https://avatars.githubusercontent.com/u/1561955?v=4',
    gravatar_id: '',
    url: 'https://api.github.com/users/midudev',
    html_url: 'https://github.com/midudev',
    followers_url: 'https://api.github.com/users/midudev/followers',
    following_url:
      'https://api.github.com/users/midudev/following{/other_user}',
    gists_url: 'https://api.github.com/users/midudev/gists{/gist_id}',
    starred_url: 'https://api.github.com/users/midudev/starred{/owner}{/repo}',
    subscriptions_url: 'https://api.github.com/users/midudev/subscriptions',
    organizations_url: 'https://api.github.com/users/midudev/orgs',
    repos_url: 'https://api.github.com/users/midudev/repos',
    events_url: 'https://api.github.com/users/midudev/events{/privacy}',
    received_events_url: 'https://api.github.com/users/midudev/received_events',
    type: 'User',
    site_admin: false
  },
  html_url: 'https://github.com/midudev/midu.dev',
  description: 'midu.dev blog 📝',
  fork: false,
  url: 'https://api.github.com/repos/midudev/midu.dev',
  forks_url: 'https://api.github.com/repos/midudev/midu.dev/forks',
  keys_url: 'https://api.github.com/repos/midudev/midu.dev/keys{/key_id}',
  collaborators_url:
    'https://api.github.com/repos/midudev/midu.dev/collaborators{/collaborator}',
  teams_url: 'https://api.github.com/repos/midudev/midu.dev/teams',
  hooks_url: 'https://api.github.com/repos/midudev/midu.dev/hooks',
  issue_events_url:
    'https://api.github.com/repos/midudev/midu.dev/issues/events{/number}',
  events_url: 'https://api.github.com/repos/midudev/midu.dev/events',
  assignees_url:
    'https://api.github.com/repos/midudev/midu.dev/assignees{/user}',
  branches_url:
    'https://api.github.com/repos/midudev/midu.dev/branches{/branch}',
  tags_url: 'https://api.github.com/repos/midudev/midu.dev/tags',
  blobs_url: 'https://api.github.com/repos/midudev/midu.dev/git/blobs{/sha}',
  git_tags_url: 'https://api.github.com/repos/midudev/midu.dev/git/tags{/sha}',
  git_refs_url: 'https://api.github.com/repos/midudev/midu.dev/git/refs{/sha}',
  trees_url: 'https://api.github.com/repos/midudev/midu.dev/git/trees{/sha}',
  statuses_url: 'https://api.github.com/repos/midudev/midu.dev/statuses/{sha}',
  languages_url: 'https://api.github.com/repos/midudev/midu.dev/languages',
  stargazers_url: 'https://api.github.com/repos/midudev/midu.dev/stargazers',
  contributors_url:
    'https://api.github.com/repos/midudev/midu.dev/contributors',
  subscribers_url: 'https://api.github.com/repos/midudev/midu.dev/subscribers',
  subscription_url:
    'https://api.github.com/repos/midudev/midu.dev/subscription',
  commits_url: 'https://api.github.com/repos/midudev/midu.dev/commits{/sha}',
  git_commits_url:
    'https://api.github.com/repos/midudev/midu.dev/git/commits{/sha}',
  comments_url:
    'https://api.github.com/repos/midudev/midu.dev/comments{/number}',
  issue_comment_url:
    'https://api.github.com/repos/midudev/midu.dev/issues/comments{/number}',
  contents_url:
    'https://api.github.com/repos/midudev/midu.dev/contents/{+path}',
  compare_url:
    'https://api.github.com/repos/midudev/midu.dev/compare/{base}...{head}',
  merges_url: 'https://api.github.com/repos/midudev/midu.dev/merges',
  archive_url:
    'https://api.github.com/repos/midudev/midu.dev/{archive_format}{/ref}',
  downloads_url: 'https://api.github.com/repos/midudev/midu.dev/downloads',
  issues_url: 'https://api.github.com/repos/midudev/midu.dev/issues{/number}',
  pulls_url: 'https://api.github.com/repos/midudev/midu.dev/pulls{/number}',
  milestones_url:
    'https://api.github.com/repos/midudev/midu.dev/milestones{/number}',
  notifications_url:
    'https://api.github.com/repos/midudev/midu.dev/notifications{?since,all,participating}',
  labels_url: 'https://api.github.com/repos/midudev/midu.dev/labels{/name}',
  releases_url: 'https://api.github.com/repos/midudev/midu.dev/releases{/id}',
  deployments_url: 'https://api.github.com/repos/midudev/midu.dev/deployments',
  created_at: '2018-11-20T21:29:52Z',
  updated_at: '2022-01-22T21:11:15Z',
  pushed_at: '2022-01-22T07:02:23Z',
  git_url: 'git://github.com/midudev/midu.dev.git',
  ssh_url: 'git@github.com:midudev/midu.dev.git',
  clone_url: 'https://github.com/midudev/midu.dev.git',
  svn_url: 'https://github.com/midudev/midu.dev',
  homepage: 'https://midu.dev',
  size: 36688,
  stargazers_count: 93,
  watchers_count: 93,
  language: 'html',
  has_issues: true,
  has_projects: false,
  has_downloads: true,
  has_wiki: false,
  has_pages: false,
  forks_count: 79,
  mirror_url: null,
  archived: false,
  disabled: false,
  open_issues_count: 5,
  license: {
    key: 'gpl-3.0',
    name: 'GNU General Public License v3.0',
    spdx_id: 'GPL-3.0',
    url: 'https://api.github.com/licenses/gpl-3.0',
    node_id: 'MDc6TGljZW5zZTk='
  },
  allow_forking: true,
  is_template: false,
  topics: [
    'blog',
    'frontend',
    'frontend-articles',
    'frontend-web',
    'hacktoberfest'
  ],
  visibility: 'public',
  forks: 79,
  open_issues: 5,
  watchers: 93,
  default_branch: 'master'
}

describe('Render RepositoryItem', () => {
  it('should render RepositoryItem', () => {
    const { container } = render(<RepositoryItem repository={repositoryMock} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
