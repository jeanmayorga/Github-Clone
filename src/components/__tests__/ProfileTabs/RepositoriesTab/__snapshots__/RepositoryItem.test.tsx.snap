// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render RepositoryItem should render RepositoryItem 1`] = `
<div
  class="py-6 border-b border-[#d8dee4] dark:border-[#373e47] grid grid-cols-12"
>
  <div
    class="col-span-9"
  >
    <h3
      class="mb-1 text-xl break-all text-[#0969da] dark:text-[#539bf5] font-semibold"
    >
      midu.dev
      <span
        class="ml-2 inline-block px-[7px] text-xs font-medium border border-[#d0d7de] text-[#57606a] dark:border-[#444c56] dark:text-[#768390] rounded-full"
      >
        Public
      </span>
    </h3>
    <p
      class="mb-2 text-[#768390]"
    >
      midu.dev blog üìù
    </p>
    <div
      class="my-1 mb-2"
    >
      <div
        class="inline-block text-xs px-[10px] font-medium rounded-full leading-5 bg-[#ddf4ff] text-[#0969da] dark:text-[#539bf5] dark:bg-[#4184e426] mr-1"
      >
        blog
      </div>
      <div
        class="inline-block text-xs px-[10px] font-medium rounded-full leading-5 bg-[#ddf4ff] text-[#0969da] dark:text-[#539bf5] dark:bg-[#4184e426] mr-1"
      >
        frontend
      </div>
      <div
        class="inline-block text-xs px-[10px] font-medium rounded-full leading-5 bg-[#ddf4ff] text-[#0969da] dark:text-[#539bf5] dark:bg-[#4184e426] mr-1"
      >
        frontend-articles
      </div>
      <div
        class="inline-block text-xs px-[10px] font-medium rounded-full leading-5 bg-[#ddf4ff] text-[#0969da] dark:text-[#539bf5] dark:bg-[#4184e426] mr-1"
      >
        frontend-web
      </div>
      <div
        class="inline-block text-xs px-[10px] font-medium rounded-full leading-5 bg-[#ddf4ff] text-[#0969da] dark:text-[#539bf5] dark:bg-[#4184e426] mr-1"
      >
        hacktoberfest
      </div>
    </div>
    <div
      class="text-[#768390] text-xs"
    >
      <span
        class="mr-4"
      >
        <span
          class="relative top-[1px] inline-block w-[12px] h-[12px] border border-[#cdd9e533] rounded-full"
          style="background-color: rgb(227, 76, 38);"
        />
         
        html
      </span>
      <span
        class="mr-4"
      >
        <svg
          aria-label="star"
          class="inline-block overflow-visible align-text-bottom  fill-[#768390]"
          data-view-component="true"
          height="16"
          role="img"
          version="1.1"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M8 .25a.75.75 0 01.673.418l1.882 3.815 4.21.612a.75.75 0 01.416 1.279l-3.046 2.97.719 4.192a.75.75 0 01-1.088.791L8 12.347l-3.766 1.98a.75.75 0 01-1.088-.79l.72-4.194L.818 6.374a.75.75 0 01.416-1.28l4.21-.611L7.327.668A.75.75 0 018 .25zm0 2.445L6.615 5.5a.75.75 0 01-.564.41l-3.097.45 2.24 2.184a.75.75 0 01.216.664l-.528 3.084 2.769-1.456a.75.75 0 01.698 0l2.77 1.456-.53-3.084a.75.75 0 01.216-.664l2.24-2.183-3.096-.45a.75.75 0 01-.564-.41L8 2.694v.001z"
            fill-rule="evenodd"
          />
        </svg>
         
        93
      </span>
      <span
        class="mr-4"
      >
        Updated 
        18 hours ago
      </span>
    </div>
  </div>
  <div
    class="col-span-3 text-right"
  >
    <div
      class="inline-flex"
    >
      <button
        class="rounded-r-none border-[#1b1f2426] hover:border-[#1b1f2426] dark:border-[#cdd9e51a] dark:hover:border-[#768390] text-[#24292f] dark:text-[#adbac7] bg-[#f6f8fa] hover:bg-[#f3f4f6] active:bg-[#ebecf0] dark:bg-[#373e47] dark:hover:bg-[#444c56] dark:active:bg-[#3d444d] w-full relative text-sm py-[5px] px-[16px] text-center font-medium whitespace-nowrap cursor-pointer align-middle select-none rounded-md border transition-all disabled:bg-[#f6f8fa] disabled:cursor-default disabled:border-[#1b1f2426]"
      >
        <svg
          aria-hidden="true"
          class="mr-2 icon fill-[#6e7781]"
          data-view-component="true"
          height="16"
          version="1.1"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M8 .25a.75.75 0 01.673.418l1.882 3.815 4.21.612a.75.75 0 01.416 1.279l-3.046 2.97.719 4.192a.75.75 0 01-1.088.791L8 12.347l-3.766 1.98a.75.75 0 01-1.088-.79l.72-4.194L.818 6.374a.75.75 0 01.416-1.28l4.21-.611L7.327.668A.75.75 0 018 .25zm0 2.445L6.615 5.5a.75.75 0 01-.564.41l-3.097.45 2.24 2.184a.75.75 0 01.216.664l-.528 3.084 2.769-1.456a.75.75 0 01.698 0l2.77 1.456-.53-3.084a.75.75 0 01.216-.664l2.24-2.183-3.096-.45a.75.75 0 01-.564-.41L8 2.694v.001z"
            fill-rule="evenodd"
          />
        </svg>
        Star
      </button>
      <button
        class="rounded-l-none border-[#1b1f2426] hover:border-[#1b1f2426] dark:border-[#cdd9e51a] dark:hover:border-[#768390] text-[#24292f] dark:text-[#adbac7] bg-[#f6f8fa] hover:bg-[#f3f4f6] active:bg-[#ebecf0] dark:bg-[#373e47] dark:hover:bg-[#444c56] dark:active:bg-[#3d444d] w-full relative text-sm py-[5px] px-[16px] text-center font-medium whitespace-nowrap cursor-pointer align-middle select-none rounded-md border transition-all disabled:bg-[#f6f8fa] disabled:cursor-default disabled:border-[#1b1f2426]"
      >
        <span
          class="inline-block align-middle content-[\\"\\"] border-4 border-b-0 border-r-transparent border-b-transparent border-l-transparent opacity-80"
        />
      </button>
    </div>
  </div>
</div>
`;
